pkg/image/trigger/deploymentconfigs/deploymentconfigs.go

```diff
@@ -0,0 +1,253 @@
+		fieldPath := fmt.Sprintf("spec.template.spec.initContainers[@name='%s'].image", dc.Spec.Template.Spec.Containers[index].Name)
+		triggers = append(triggers, triggerapi.ObjectFieldTrigger{
+			From: triggerapi.ObjectReference{
+				Name:       from.Name,
+				Namespace:  from.Namespace,
+				Kind:       from.Kind,
+				APIVersion: from.APIVersion,
+			},
+			FieldPath: fieldPath,
+			Paused:    !t.ImageChangeParams.Automatic,
+		})
+	}
+	return triggers
+}
+
+// calculateDeploymentConfigTriggers transforms a deployment config into a set of image change triggers. If retrieveChanges
+// is true an array of the latest state of the triggers will be returned.
+func calculateDeploymentConfigTriggers(dc *deployapi.DeploymentConfig) []triggerapi.ObjectFieldTrigger {
+	var triggers []triggerapi.ObjectFieldTrigger
+	for _, t := range dc.Spec.Triggers {
+		addedTriggers := calculateDeploymentConfigTrigger(t, dc)
+		triggers = append(triggers, addedTriggers...)
+	}
+	return triggers
+}
+
+// deploymentConfigTriggerIndexer converts deployment config events into entries for the trigger cache, and
+// also calculates the latest state of the changes on the object.
+type deploymentConfigTriggerIndexer struct {
+	prefix string
+}
+
+func NewDeploymentConfigTriggerIndexer(prefix string) trigger.Indexer {
+	return deploymentConfigTriggerIndexer{prefix: prefix}
+}
+
+func (i deploymentConfigTriggerIndexer) Index(obj, old interface{}) (string, *trigger.CacheEntry, cache.DeltaType, error) {
+	var (
+		triggers []triggerapi.ObjectFieldTrigger
+		dc       *deployapi.DeploymentConfig
+		change   cache.DeltaType
+	)
+	switch {
+	case obj != nil && old == nil:
+		// added
+		dc = obj.(*deployapi.DeploymentConfig)
+		triggers = calculateDeploymentConfigTriggers(dc)
+		change = cache.Added
+	case old != nil && obj == nil:
+		// deleted
+		dc = old.(*deployapi.DeploymentConfig)
+		triggers = calculateDeploymentConfigTriggers(dc)
+		change = cache.Deleted
+	default:
+		// updated
+		dc = obj.(*deployapi.DeploymentConfig)
+		triggers = calculateDeploymentConfigTriggers(dc)
+		oldTriggers := calculateDeploymentConfigTriggers(old.(*deployapi.DeploymentConfig))
+		switch {
+		case len(oldTriggers) == 0:
+			change = cache.Added
+		case !reflect.DeepEqual(oldTriggers, triggers):
+			change = cache.Updated
+		}
+	}
+
+	if len(triggers) > 0 {
+		key := i.prefix + dc.Namespace + "/" + dc.Name
+		return key, &trigger.CacheEntry{
+			Key:       key,
+			Namespace: dc.Namespace,
+			Triggers:  triggers,
+		}, change, nil
+	}
+	return "", nil, change, nil
+}
+
+// DeploymentConfigReactor converts image trigger changes into updates on deployments.
+type DeploymentConfigReactor struct {
+	Client client.DeploymentConfigsNamespacer
+}
+
+func UpdateDeploymentConfigImages(dc *deployapi.DeploymentConfig, tagRetriever trigger.TagRetriever) (*deployapi.DeploymentConfig, bool, error) {
```

Comment:
There is overlap between UpdateDeploymentConfigImages and [processTriggers](https://github.com/openshift/origin/blob/b38274c5227320dc336b707ea766e96a194261fe/pkg/deploy/registry/instantiate/rest.go#L112).